### PR TITLE
[Blocks editor] Fix- The first time you click on the List option or button the list is not created

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

It fixes the problem we had when we tried to create a new list, the first time you click on the button or option it doesn't work

### Why is it needed?

To solve the issue

### How to test it?

Create a new instance with a Block Field and then click directly on the list Button, you need to see the list created (same for the list option)

### CODE REVIEW

**Review by joshuaellis — COMMENTED** [review]
All your comments are better, but really you're describing what I can read e.g. remove a node etc.

What you're really missing is the why behind it's doing what it's doing... but great to see nonetheless!

**Comment by simotae14** [comment]
@joshuaellis let me know if it is better now, thanks

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js [hunk 1]
```diff
@@ -614,6 +636,118 @@ describe('BlocksEditor toolbar', () => {
...
+            ],
+          },
+        ],
+      },
+    ]);
+
+    expect(ReactEditor.focus).toHaveBeenCalledTimes(1);
+  });
+
+  it('creates a new list with some content when you select the option in the dropdown and editor contains an heading on the last line', async () => {
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:712]:
```suggestion
  it('creates a new list with some content when you select the option in the dropdown and editor contains a heading on the last line', async () => {
```

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js [hunk 2]
```diff
@@ -614,6 +636,118 @@ describe('BlocksEditor toolbar', () => {
...
+                type: 'text',
+                text: 'A heading two',
+              },
+            ],
+          },
+        ],
+      },
+    ]);
+  });
   it('should disable the modifiers buttons when the selection is inside an image', async () => {
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:751]: Please always add blank lines between tests

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js [hunk 3]
```diff
@@ -48,6 +48,28 @@ const mixedInitialValue = [
   },
 ];
 
+const paragraphWithModifiersInitialValue = [
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:51]: This seems to only be used once, you can inline it in the setup()

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 4]
```diff
@@ -464,21 +464,58 @@ const isListActive = (editor, matchNode) => {
...
+      {
+        at: [lastNodePath[0]],
+        select: true,
+      }
+    );
+
+    if (!isActive) {
+      const block = { type: 'list', format, children: [] };
+      Transforms.wrapNodes(editor, block);
+    }
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:518]: Could you keep the old comment here?

>If the selection is now a list item, wrap it inside a list

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 5]
```diff
@@ -464,21 +464,58 @@ const isListActive = (editor, matchNode) => {
...
+    /**
+     * Otherwise if there is not a selected portion in the editor, we want to convert the last inserted node to a list or if it is already a list, convert it back to a paragraph
+     */
+  } else {
+    const [, lastNodePath] = Editor.last(editor, []);
+
+    const [parentNode] = Editor.parent(editor, lastNodePath, {
+      match: (node) => node.type !== 'text', // makes sure we get a block node, not an inline node
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:491]: Please move the comment to its own line avoid having too long lines

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 6]
```diff
@@ -464,21 +464,58 @@ const isListActive = (editor, matchNode) => {
...
+    });
+
+    if (!isActive) {
+      const block = { type: 'list', format, children: [] };
+      Transforms.wrapNodes(editor, block);
+    }
+    /**
+     * Otherwise if there is not a selected portion in the editor, we want to convert the last inserted node to a list or if it is already a list, convert it back to a paragraph
+     */
+  } else {
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:487]: can you move the comment inside of the else block?

also, please avoid having lines of comments that are longer than the lines of code, as they're harder to read in an IDE. For this one you could split it like this for example:

```js
// There is no selection, convert the last inserted node to a list
// If it is already a list, convert it back to a paragraph
```

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 7]
```diff
@@ -464,21 +464,58 @@ const isListActive = (editor, matchNode) => {
...
 const toggleList = (editor, isActive, format) => {
-  // Delete the parent list so that we're left with only the list items directly
-  Transforms.unwrapNodes(editor, {
-    match: (node) => isListNode(node) && ['ordered', 'unordered'].includes(node.format),
-    split: true,
-  });
+  /**
+   * If we have selected a portion of content in the editor, we want to convert it to a list or if it is already a list, convert it back to a paragraph
+   */
+  if (editor.selection) {
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:470]: Same as my other comment, please split into several comment lines

**Review by remidej — COMMENTED** [review]
Haven't done QA yet, just code formatting comments

**Review by remidej — APPROVED** [review]

**Pull request merged**